ifrs17/Report/ReportMutableScopesInteractive.ipynb

```diff
@@ -0,0 +1,529 @@
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "### Monthly Period"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "[InitializeScope(nameof(InitReportingPeriod))]",
+                "\npublic interface MonthlyPeriodFormsEntity : MutableScopeWithWorkspace {",
+                "\n    [DropdownMethod(nameof(GetReportingPeriodAutocompleteAsync))]",
+                "\n    string ReportingPeriod { get; set; }",
+                "\n",
+                "\n    private char separator => 'M';",
+                "\n    private string[] ReportingPeriodSplit => ReportingPeriod.Split(separator);",
+                "\n    private int ParseReportingPeriod(int index) => !string.IsNullOrWhiteSpace(ReportingPeriod) && ReportingPeriodSplit is not null && Int32.TryParse(ReportingPeriodSplit.ElementAtOrDefault(index), out int value)",
+                "\n        ? value",
+                "\n        : default;",
+                "\n",
+                "\n    protected int Year => ParseReportingPeriod(0);",
+                "\n    protected int Month => ParseReportingPeriod(1);",
+                "\n",
+                "\n    async Task<IReadOnlyCollection<string>> GetReportingPeriodAutocompleteAsync(string userInput, int page, int pageSize) => ",
+                "\n        await workspace.Query<PartitionByReportingNodeAndPeriod>()",
+                "\n            .Where(x => x.Scenario == null)",
+                "\n            .OrderByDescending(x => x.Year)",
+                "\n            .ThenByDescending(x => x.Month)",
+                "\n            .Select(x => ParseReportingPeriodToDisplayString(x.Year, x.Month, separator))",
+                "\n            .Where(x => userInput == null || x.Contains(userInput, StringComparison.OrdinalIgnoreCase))",
+                "\n            .ToArrayAsync();",
+                "\n",
+                "\n    async void InitReportingPeriod() {",
+                "\n        ReportingPeriod = ParseReportingPeriodToDisplayString(GetStorage().InitialReportingPeriod.Year, GetStorage().InitialReportingPeriod.Month, separator);",
+                "\n    }",
+                "\n}"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "markdown",
+            "source": [
+                "## Filters"
+            ],
+            "metadata": {},
+            "execution_count": 0,
+            "outputs": []
+        },
+        {
+            "cell_type": "code",
+            "source": [
+                "public interface BasicFilterFormsEntity : MutableScopeWithWorkspace {",
+                "\n    [DropdownValues(\"\", \"GroupOfContract\", \"Novelty\", \"EconomicBasis\")]",
```

Comment:
use nameof instead of hardcoding strings.

Reply:
The list of dropdown values is not complete. E.g. AmountType is missing. 
However AmountType (and EconomicBasis) is not relevant for some report (see csm). 
How would we proceed in that case? Have a new scope inheriting from this one changing the attribute of FilterName?

Reply:
This is not finished